app/javascript/packs/rules.js

```diff
@@ -0,0 +1,64 @@
+import TurbolinksAdapter from 'vue-turbolinks'
+import Vue from 'vue'
+import {ButtonPlugin, BreadcrumbPlugin, FormPlugin, FormGroupPlugin, FormTextareaPlugin, BadgePlugin, CollapsePlugin} from 'bootstrap-vue'
+import Rules from '../components/rules/Rules.vue'
+import RulesCodeEditorView from '../components/rules/RulesCodeEditorView.vue'
+import RuleComments from '../components/rules/RuleComments.vue'
+import RuleHistories from '../components/rules/RuleHistories.vue'
+import RuleNavigator from '../components/rules/RuleNavigator.vue'
+
+Vue.use(TurbolinksAdapter)
+Vue.use(ButtonPlugin)
+Vue.use(BreadcrumbPlugin)
+Vue.use(FormPlugin)
+Vue.use(FormGroupPlugin)
+Vue.use(FormTextareaPlugin)
+Vue.use(BadgePlugin)
+Vue.use(CollapsePlugin)
+
+Vue.component('Rules', Rules)
+Vue.component('RulesCodeEditorView', RulesCodeEditorView)
+Vue.component('RuleComments', RuleComments)
+Vue.component('RuleNavigator', RuleNavigator)
+Vue.component('RuleHistories', RuleHistories)
+
+// Based on what the server response is, account for
+// generating an alert or notice on the user's screen
+Vue.prototype.alertOrNotifyResponse = function(response) {
+  let classes = 'alert alert-dismissable fade show ';
+  let textContent = '';
+  if (response['data'] && response['data']['notice']) {
+    classes += ' alert-success';
+    textContent = response['data']['notice'];
+  }
+  else if (response['data'] && response['data']['alert']) {
+    classes += 'alert-danger';
+    textContent = response['data']['alert'];
+  } else {
+    // The response did not contain data we can use for an alert or notice.
+    return;
+  }
+
+  let element = document.createElement('p');
+  element.className = classes;
+  element.textContent = textContent;
+  element.setAttribute('role', 'alert');
+  navbar?.insertAdjacentElement('afterend', element);
+}
+
+// Format a date time string like '2021-08-10T19:43:24.950Z'
+// into a more readable format.
+Vue.prototype.friendlyDateTime = function(dateTimeString) {
+  const date = new Date(dateTimeString);
+  const hours = date.getHours();
+  const amOrPm = hours < 12 ? ' AM' : ' PM';
+  const minutes = date.getMinutes() < 10 ? "0" + date.getMinutes() : date.getMinutes()
+  const timeString = (hours > 12 ? hours - 12 : hours) + ":" + minutes + amOrPm;
+  return `${date.toDateString()} @ ${timeString}`;
+}
```

Comment:
Is there a library that handles this if we respond with integer time? Something like a call to strftime ?

Reply:
Additionally, this looks like something that should be brought up to a utility or helper style location. I could see this being reused among multiple components.

Reply:
now using moment.js and a centralized mixin for this